hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneManagerSnapshotAcl.java

```diff
@@ -409,7 +409,9 @@ private void setDefaultAcls() throws IOException {
         .build();
     objectStore.setAcl(volumeObj, OzoneAcl.parseAcls(
         "user:" + USER1 + ":r," +
-        "user:" + USER2 + ":r"));
+        "user:" + USER1 + ":l," +
```

Comment:
Why do you need ":l" permission for volume object? You have modified to handle only keys permission in ListStatus.

Reply:
@ashishkumar50 Hi, Because list the volume or bucket also requires the "l" permission
For example
```
[hadoop@bigdata-pre-master01.nmg01 ~]$ ~/om-current/bin/ozone sh  vol create vol-test
[hadoop@bigdata-pre-master01.nmg01 ~]$ ~/om-current/bin/ozone sh  bucket create vol-test/blk-test

[hadoop@bigdata-pre-hdp02.nmg01 ~]$ export HADOOP_USER_NAME=test
[hadoop@bigdata-pre-hdp02.nmg01 ~]$  ~/hadoop-2.7.2-5504-ozone-client/bin/hadoop fs -ls /
[hadoop@bigdata-pre-hdp02.nmg01 ~]$  ~/hadoop-2.7.2-5504-ozone-client/bin/hadoop fs -ls /vol-test
-ls: Fatal internal error
java.lang.RuntimeException: PERMISSION_DENIED org.apache.hadoop.ozone.om.exceptions.OMException: User test doesn't have LIST permission to access volume Volume:vol-test
	at org.apache.hadoop.ozone.client.OzoneVolume$BucketIterator.getNextListOfBuckets(OzoneVolume.java:527)
	at org.apache.hadoop.ozone.client.OzoneVolume$BucketIterator.<init>(OzoneVolume.java:497)
	at org.apache.hadoop.ozone.client.OzoneVolume.listBuckets(OzoneVolume.java:368)
	at org.apache.hadoop.fs.ozone.BasicRootedOzoneClientAdapterImpl.listStatusVolume(BasicRootedOzoneClientAdapterImpl.java:801)
	at org.apache.hadoop.fs.ozone.BasicRootedOzoneClientAdapterImpl.listStatus(BasicRootedOzoneClientAdapterImpl.java:885)
	at org.apache.hadoop.fs.ozone.BasicRootedOzoneFileSystem.listStatusAdapter(BasicRootedOzoneFileSystem.java:869)
	at org.apache.hadoop.fs.ozone.BasicRootedOzoneFileSystem.listStatus(BasicRootedOzoneFileSystem.java:844)
	at org.apache.hadoop.fs.shell.PathData.getDirectoryContents(PathData.java:268)
	at org.apache.hadoop.fs.shell.Command.recursePath(Command.java:373)
	at org.apache.hadoop.fs.shell.Ls.processPathArgument(Ls.java:90)
	at org.apache.hadoop.fs.shell.Command.processArgument(Command.java:271)
	at org.apache.hadoop.fs.shell.Command.processArguments(Command.java:255)
	at org.apache.hadoop.fs.shell.Command.processRawArguments(Command.java:201)
	at org.apache.hadoop.fs.shell.Command.run(Command.java:165)
	at org.apache.hadoop.fs.FsShell.run(FsShell.java:287)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:70)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:84)
	at org.apache.hadoop.fs.FsShell.main(FsShell.java:340)
Caused by: PERMISSION_DENIED org.apache.hadoop.ozone.om.exceptions.OMException: User test doesn't have LIST permission to access volume Volume:vol-test
	at org.apache.hadoop.ozone.om.protocolPB.OzoneManagerProtocolClientSideTranslatorPB.handleError(OzoneManagerProtocolClientSideTranslatorPB.java:710)
	at org.apache.hadoop.ozone.om.protocolPB.OzoneManagerProtocolClientSideTranslatorPB.listBuckets(OzoneManagerProtocolClientSideTranslatorPB.java:632)
	at org.apache.hadoop.ozone.client.rpc.RpcClient.listBuckets(RpcClient.java:1188)
	at org.apache.hadoop.ozone.client.OzoneVolume$BucketIterator.getNextListOfBuckets(OzoneVolume.java:525)
	... 17 more
```

Reply:
But according to documentation it needs only READ permission on volume and for bucket it needs LIST and READ for listKeys. Can you please check for list Key behaviour as well.
https://ozone.apache.org/docs/1.3.0/security/securitywithranger.html